app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts

```diff
@@ -196,6 +198,44 @@ const isNoSchemaAvailable = (schema) => {
   return true;
 };
 
+const parseImportedSchemas = (schemas) => {
+  let importedSchemas = schemas;
+  if (typeof schemas === 'string') {
+    importedSchemas = JSON.parse(schemas);
+  }
+
+  if (!Array.isArray(importedSchemas)) {
+    // This will be the case when we hit 'Apply' button from wrangler.
+    // We don't maintain consistency in importing schemas from multiple
+    // sources. This will take time to fix as we right now throw schemas
+    // in multiple formats across all places.
+    if (isObject(importedSchemas) && !importedSchemas.hasOwnProperty('schema')) {
```

Comment:
why the mix of isObject and typeof 'object'?

Reply:
Copied from the existing code. Since this will be cherry-picked, I didn't want to make any unnecessary changes here.